lib/auth/integration/integrationv1/service.go

```diff
@@ -230,6 +231,13 @@ func (s *Service) CreateIntegration(ctx context.Context, req *integrationpb.Crea
 		if err := s.createGitHubCredentials(ctx, req.Integration); err != nil {
 			return nil, trace.Wrap(err)
 		}
+	case types.IntegrationSubKindAWSOIDC:
+		// AWS OIDC Integration can be used as source of credentials to access AWS Web/CLI.
+		// This creates a new AppServer whose endpoint is <integrationName>.<proxyURL>, which can fail if integrationName is not a valid DNS Label.
+		// Instead of failing when the integration is already created, it fails at creation time.
+		if len(validation.IsDNS1035Label(req.GetIntegration().GetName())) > 0 {
```

Comment:
Nit: This function's usage is slightly confusing, I think this way it will be clearer that it returns errors.
```suggestion
		if errs := validation.IsDNS1035Label(req.GetIntegration().GetName()); len(errs) > 0 {
```